File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.py
def solution(my_str, n):
    # 잘라진 문자열을 저장할 리스트 초기화
    answer = []

    # my_str이 남아있는 동안 반복
    while my_str:
        # 앞에서부터 n개의 문자 슬라이싱하여 리스트에 추가
        answer.append(my_str[:n])
        # n개만큼 잘라낸 후 나머지 문자열로 업데이트
        my_str = my_str[n:]

    # 최종적으로 분할된 문자열 리스트 반환
    return answer

# ✅ 예제 테스트 실행
print(solution("abc1Addfggg4556b", 6))
# 결과: ['abc1Ad', 'dfggg4', '556b'] (6글자씩 나눠서 저장)

print(solution("abcdef123", 3))
# 결과: ['abc', 'def', '123'] (3글자씩 나눠서 저장)

File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.java
import java.util.*;

public class Solution {

    public static String[] solution(String my_str, int n) {
        List<String> result = new ArrayList<>();

        int length = my_str.length();
        for (int i = 0; i < length; i += n) {
            // i부터 n글자 잘라내되, 범위를 초과하면 끝까지
            int end = Math.min(i + n, length);
            result.add(my_str.substring(i, end));
        }

        // List<String> → String[]
        return result.toArray(new String[0]);
    }

    // ✅ 테스트 실행
    public static void main(String[] args) {
        System.out.println(Arrays.toString(solution("abc1Addfggg4556b", 6)));
        // 출력: [abc1Ad, dfggg4, 556b]

        System.out.println(Arrays.toString(solution("abcdef123", 3)));
        // 출력: [abc, def, 123]
    }
}
